### ISSUE TITLE
Bug: LanguageModel batched trace silently mispredicts padded prompts on absolute-position models (GPT-2)

### ISSUE THREAD

**Issue by khaiwang** [ndif-team/nnsight#672]
A multi-prompt `model.trace([p1, p2, …])` on a model with **learned absolute position embeddings** (GPT-2 family) returns silently wrong per-prompt activations and logits for **every prompt shorter than the longest one in the batch**. No error is raised.

## Root cause

`LanguageModel` left-pads batches (`padding_side="left"`) and forwards `input_ids` + `attention_mask` but never `position_ids`. transformers' GPT-2 then defaults `position_ids` to a mask-blind `arange` (`modeling_gpt2.py`), so a padded row's real tokens get absolute positions shifted by the pad count. HF expects the caller to pass `position_ids` for left-padded batches — `generate()` does, a bare `forward()` does not — so the gap is on the nnsight side: it left-pads and calls the bare forward, with the mask in hand, but omits the `position_ids` correction.

## Severity (measured, GPT-2-small)

Not gradual — a *single* pad token already moves the next-token distribution ~0.9 total-variation (of a 1.0 max) and flips the top-1 for most prompts; **≥2 pad tokens flip 100%**. It saturates immediately because the learned position embeddings for adjacent indices are arbitrary, not-close vectors, so a one-position shift injects a large perturbation at the residual-stream input that the LayerNorms don't normalize away. Only the longest (unpadded) row is correct. It corrupts *all* activations of the padded rows, so every methodology (logit lens, steering, ablation, patching, caching) is affected, not just final logits.

| pad k | mean TV (next-token) | top-1 flip rate | final-hidden cos |
|--:|--:|--:|--:|
| 0   | 0.000 | 0%   | 1.000 |
| 1   | 0.897 | 83%  | 0.912 |
| 2   | 0.875 | 100% | 0.928 |
| 8   | 0.912 | 100% | 0.795 |
| 128 | 0.900 | 100% | 0.509 |

## Scope

- **Affected:** learned-absolute-position models (GPT-2, GPT-Neo), the `.trace()` forward path, mixed-length batches. (Some learned-absolute models — e.g. OPT, RoBERTa — derive `position_ids` from the mask in their own forward and are safe; the precise criterion is "does the forward use the raw absolute index, mask-blind," which GPT-2 does.)
- **Unaffected:** RoPE / ALiBi (relative) models — attention depends only on the token offset, which a uniform left-pad shift preserves; single-prompt traces (no padding); `.generate()` (HF supplies `position_ids`); and the vLLM backend (each prompt runs as its own unpadded request).

## Repro

```python
import nnsight
m = nnsight.LanguageModel("openai-community/gpt2", dispatch=True)
p = ["Hello world", "The Eiffel Tower is located in the city of"]   # different lengths

with m.trace(p):
    batched = m.lm_head.output[:, -1].save()
with m.trace(p[0]):
    alone = m.lm_head.output[:, -1].save()

print((batched[0] - alone[0]).abs().max())   # ~16 on main; ~0 with the fix
```

## Fix

Derive `position_ids` from the attention mask in `LanguageModel._prepare_input` / `_batch` (as `generate()` does). See the linked PR.
